core/controller/src/main/scala/whisk/core/controller/Meta.scala

```diff
@@ -98,14 +190,21 @@ protected[core] object WhiskMetaApi extends Directives {
         extension: String,
         defaultProjection: Option[List[String]],
         projectionAllowed: Boolean,
-        transcoder: (JsValue, TransactionId) => RequestContext => Unit)
+        transcoder: (JsValue, TransactionId, WebApiDirectives) => RequestContext => Unit) {
+        val extensionLength = extension.length
+    }
 
-    private def resultAsHtml(result: JsValue, transid: TransactionId): RequestContext => Unit = result match {
+    private def resultAsHtml(result: JsValue, transid: TransactionId, rp: WebApiDirectives): RequestContext => Unit = result match {
```

Comment:
fair question - and i tried to find a convenient workaround. we need to access the status code directives inside the response as http handler to determine the value of the "status code" field. the signature for all of these methods is the same, so for that reason, the directive is provided to all the methods. alternatively, one could create another closure with the status code field captured which seemed more complex and unnecessary.